decidim-admin/app/commands/decidim/admin/destroy_area.rb

```diff
@@ -20,14 +20,22 @@ def initialize(area, current_user)
       #
       # Returns nothing.
       def call
-        destroy_area
-        broadcast(:ok)
+        if area_has_dependencies?
+          broadcast(:has_spaces)
+        else
+          destroy_area
+          broadcast(:ok)
+        end
       end
 
       private
 
       attr_reader :current_user
 
+      def area_has_dependencies?
+        ::Decidim::ParticipatoryProcess.where(area: @area).exists? || ::Decidim::Assembly.where(area: @area).exists?
```

Comment:
I'm not sure that this is correct ...
What would happen if one of the two space modules (`Decidim::ParticipatoryProcess` or `Decidim::Assembly`) is not installed?
Also how `Scope` are checking if has_dependencies?

Reply:
Hi @isaacmg410 I've refactored the method to use the participatory spaces registry so that this references are now abstracted. :smile: 
Thanks for pointing out! :+1: